enhancements/SEP2-revision-tags.md

```diff
@@ -0,0 +1,55 @@
+
+Here's an example how to create a `default` revision tag that points to the `1-21-1` revision using `istioctl`:
+
+```bash
+istioctl tag set default --revision 1-21-1
+```
+
+## Goals
+* Provide revision tag support in sail-operator APIs so users don't have to use istioctl for basic revision tag operations
+
+## Non-goals
+* Compatibility with manual revision tag creation using istioctl. There might be a way to disable the operator functionality to avoid conflicts when creating revision tags manually, but that's it - you either do it yourself or let the operator do it
+
+## Design
+
+### User Stories
+1. As a user of sail-operator's RevisionBased update strategy, I want to be able to use the `istio-injection=enabled` label on my namespaces.
+1. As a platform engineer, I want my application teams to be able to use a fixed label for proxy injection without having to know which version of Istio is running in the cluster, so that I can perform upgrades in the background without the application teams having to make changes to use the new version.
+
+### API Changes
+We will add a new field `revisionTags` of type `[]string` to the `Istio` CRD. It will be located under `spec.updateStrategy.revisionTags`. Whenever the sail-operator deploys a new revision, it will update all the tags listed in the `revisionTags` field to point to this new revision. The default value for the field will be `{"default"}` - this way, the standard namespace injection label `istio-injection=enabled` will work out of the box for RevisionBased deployments (see second paragraph of the [Overview](#overview)).
+
+We will also need to remove the `values.revisionTags` field (which is how the upstream charts expose this feature). We can still set that field to trigger creation of the tags, but we should not expose it to users.
+
+### Architecture
+We will need to update the sail-operators mechanism to detect revisions that are being used. Today, we only look at the `istio.io/rev` label's value to check which revisions are in use. But when revision tags are used, those values will be mere aliases, so we have to improve our detection mechanism. The most correct way is probably to look at the revision annotation on the pods that is set by Istio during injection. That requires inspecting every pod in the cluster, though. Another way could be to resolve the tags - the sail-operator knows which revisions the tags point to, after all - but only if tags are exclusively managed by the sail-operator.
+
+Revision tags must never overlap between `Istio` resources, as every revision tag must point to exactly one revision, and if multiple `Istio` resources define the same revision tags, they would flip-flop between the respective revisions, depending on which resource was reconciled last. We'll need to make sure during reconciliation that they are unique and report an error in the status if they are not.
+
+## Alternatives Considered
```

Comment:
What about introducing a new resource kind (e.g. `IstioRevisionTag`), which contains a reference to an `IstioRevision`? One benefit of this approach is that a tag can never point to more than one revision and there's an easy way to list all tags in the cluster. One downside is that users will have to create a default `IstioRevisionTag` to enable the use of the `istio-injection: enabled` label (this is in contrast to the current proposal, where `spec.revisionTags` defaults to `["default"]`).

Reply:
After thinking more about the `IstioRevisionTag` approach, I realized that both revisions and tags are referenced via the same `istio.io/rev` label, which means that their names must be unique across both revisions and tags (i.e. you can't have a revision called `foo` and a tag called `foo` at the same time). 

So, to ensure uniqueness, another approach would be to introduce a new `IstioRevision` type called `Tag` (in addition to the existing `Local` and `Remote` types). This ensures that users can look up all the active revision/tag names in a single list (`kubectl get istiorevisions`). The downside of this approach is that a tag is much different than a revision, since it doesn't have `values`, `version`, or `namespace`, but does have a reference to another revision. So, maybe merging revisions and tags into a single resource isn't a good approach, even though it would ensure name uniqueness. But it probably should be listed as an alternative that we have considered.

Bottom line seems to be that we can't really ensure name uniqueness through resource names, bringing into question whether it even makes sense to introduce `IstioRevisionTag`, as proposed in my previous comment.

Reply:
> What about introducing a new resource kind (e.g. IstioRevisionTag), which contains a reference to an IstioRevision?

I like this idea. What is the lifecycle of the tag tied to? The `Istio` resource? Maybe we could have a setting on the `Istio` resource that will create a default `IstioRevisionTag` for you when enabled when the `Istio` resource gets created?

Reply:
@nrfox `istioctl` doesn't allow you to create a tag with the same name as an existing revision:
```console
$ istioctl tag set main-rev --revision=main-rev
Error: cannot create revision tag "main-rev": found existing control plane revision with same name
```

Reply:
Ah I was wrong then. Well it may be worth adding this discussion to the SEP as an alternative considered regardless. 

Wouldn't having tags as a field on the `Istio` resource run into the same problem? You could add a tag to `istio.spec.updateStrategy.revisionTags` that matched an existing `IstioRevision`.

Reply:
I really like the `IstioRevisionTag` resource approach. I think the name uniqueness is not a show-stopper: we could always have a status condition that shows an error if the name isn't unique. Your tag simply never gets created